build.sh

```diff
@@ -25,226 +26,178 @@ fi
 set -eo pipefail
 
 exit_error() {
-    echo "$*" 1>&2
-    exit 1;
+	echo "$*" 1>&2
+	exit 1;
 }
 
 usage() {
-    cat <<-EOF
-	Usage: $(basename "${BASH_SOURCE[0]}") [-hu] [-k /path/to/authorized_keys] 
-	
+	cat <<-EOF
+	Usage: $(basename "${BASH_SOURCE[0]}") [-hu] [-k /path/to/authorized_keys]
+
 	-h Show help
 	-u Build with upload to remote host
-	-k /path/to/authorized_keys (can safely ignore, another opportunity to copy in SSH keys on image boot!)
+	-k /path/to/authorized_keys (can safely ignore, another opportunity to copy
+	   in SSH keys on image boot!)
 	EOF
 }
 
 # we must be on freebsd
 what_os_am_i="$(uname)"
-if [ "${what_os_am_i}" != "FreeBSD" ]; then
-    exit_error "Please run on FreeBSD only"
+if [ "$what_os_am_i" != "FreeBSD" ]; then
+	exit_error "Please run on FreeBSD only"
 fi
 
 UPLOAD="NO"
 # get command line flags
 while getopts huk: flag
 do
-    case "${flag}" in
-        h) 
-           usage
-           exit 0
-           ;;
-        u) 
-           UPLOAD="YES"
-           ;;
-        k) 
-           AUTHKEYFILE="${OPTARG}"
-           ;;
-        *) 
-           exit_error "$(usage)"
-           ;;
-    esac
+	case "$flag" in
+	h)
+		usage
+		exit 0
+		;;
+	u)
+		UPLOAD="YES"
+		;;
+	k)
+		AUTHKEYFILE="$(realpath "$OPTARG")"
```

Comment:
Use realpath, so the existence of the file is actually verified early (and the path will work everywhere)